src/api/v1/db/oracledb/ingredients-dao.js

```diff
@@ -15,6 +17,70 @@ const ingredientsColumnNames = {
   notes: 'NOTES',
 };
 
+/**
+ * convert an output bind param name like `nameOut` to a
+ * property name like `name` by removing 'Out' at the end of the string
+ * @param {string} outBindParamName
+ * @returns {string} the property name
+ */
+const outBindParamToPropertyName = (outBindParamName) => outBindParamName.slice(0, -3);
```

Comment:
I'd suggest using regex to fetch the substring you need instead of using index to slice the string.

Reply:
Hmm. I guess the advantage of a regex is that that function can more easily handle more edge cases, like getting a bind param name that doesn't end with `Out`. My thought when writing it was that because it's a private method we have pretty strict control over when it's used, so we can basically always assumed it's called with the "right" input. In that case, I thought this solution was simpler and more readable. Thoughts?

Reply:
Just my two cents, I think using regex is easier for developers including maintainers to quickly understand which part of string should be fetched since it's self-descriptive. Counting the index of a string and slice the part we need sometimes can be "simpler", but I think it'll be harder to maintain as time goes by, and it'll be even harder if you have more similar logic like that (at least you'll need to write some input and output examples for them to let others know what should be parsed).

Reply:
That makes sense. Thoughts on using `.replace(/Out$/, "")` instead of either solution? I feel like that strikes a balance of being readable and more error resilient.

Reply:
Another benefit of using regex is that you can also validate the input. I won't use replace in this case since it will not do anything if the input is incorrect. Instead, I'll use regex pattern to match the input and fetch the part we need by groups.

Reply:
Like what you did [here](https://github.com/osu-mist/pizza-api/blob/0d1a11b94795c0bcb925d47421ba97dba9b96ece/src/api/v1/db/oracledb/doughs-dao.js#L130-L132).

Reply:
Got this updated to use a regular expression 👍